extensions/amp-content-viewer/0.1/amp-content-viewer.js

```diff
@@ -0,0 +1,696 @@
+    const elementBoxRatio = this.elementBox_.width /
+     this.elementBox_.height;
+    const maxScale = Math.max(
+        elementBoxRatio / sourceAspectRatio,
+        sourceAspectRatio / elementBoxRatio
+    );
+    this.maxScale_ = Math.max(DEFAULT_MAX_SCALE, maxScale);
+
+    // Reset zoom and pan.
+    this.startScale_ = this.scale_ = 1;
+    this.startX_ = this.posX_ = 0;
+    this.startY_ = this.posY_ = 0;
+    this.updatePanZoomBounds_(this.scale_);
+  }
+
+  /**
+   * Measures and resets the content dimensions, after the element
+   * dimensions changes.
+   * @return {!Promise}
+   */
+  resetContentDimensions_() {
+    return this.measureElement(() => this.measure_()).then(() => {
+      const content = dev().assertElement(this.content_);
+      return this.mutateElement(() => {
+        // Set the actual dimensions of the content
+        st.setStyles(content, {
+          top: st.px(this.contentBox_.top),
+          left: st.px(this.contentBox_.left),
+          width: st.px(this.contentBox_.width),
+          height: st.px(this.contentBox_.height),
+        });
+        // Update translation and scaling
+        this.updatePanZoom_();
+      }, content);
+    });
+  }
+
+  /** @private */
+  cleanupGestures_() {
+    if (this.gestures_) {
+      this.gestures_.cleanup();
+      this.gestures_ = null;
+    }
+  }
+
+  /** @private */
+  setupGestures_() {
+    // TODO (#12881): this and the subsequent use of event.preventDefault
+    // is a temporary solution to #12362. We should revisit this problem after
+    // resolving #12881 or change the use of window.event to the specific event
+    // triggering the gesture.
+    this.gestures_ = Gestures.get(
+        this.element,
+        /* opt_shouldNotPreventDefault */ false
+    );
+
+    this.gestures_.onPointerDown(() => {
+      if (this.motion_) {
+        this.motion_.halt();
+      }
+    });
+
+    // Zoomable.
+    this.gestures_.onGesture(DoubletapRecognizer, e => {
+      const newScale = this.scale_ == 1 ? this.maxScale_ : this.minScale_;
+      const deltaX = this.elementBox_.width / 2 - e.data.clientX;
+      const deltaY = this.elementBox_.height / 2 - e.data.clientY;
+      this.onZoom_(newScale, deltaX, deltaY, true).then(() => {
+        return this.onZoomRelease_();
+      });
+    });
+
+    this.gestures_.onGesture(TapzoomRecognizer, e => {
+      this.onTapZoom_(e.data.centerClientX, e.data.centerClientY,
+          e.data.deltaX, e.data.deltaY);
+      if (e.data.last) {
+        this.onTapZoomRelease_(e.data.centerClientX, e.data.centerClientY,
+            e.data.deltaX, e.data.deltaY, e.data.velocityY, e.data.velocityY);
+      }
+    });
+
+    this.gestures_.onGesture(PinchRecognizer, e => {
+      this.onPinchZoom_(e.data.centerClientX, e.data.centerClientY,
+          e.data.deltaX, e.data.deltaY, e.data.dir);
+      if (e.data.last) {
+        this.onZoomRelease_();
+      }
+    });
+
+    // Override all taps
```

Comment:
Added a hack here to override all taps, so that tapping does not get triggered on double tap. This results in a UX of delaying tap effects by 300 ms.

Reply:
put this in the code comments?